File: aula-01/js/Ex01.js
/**
 * Verifica se n satisfaz a propriedade 3025.
 * @param {Number} n variavel a ser verificada
 * @throws {Error} caso entrada seja null ou undefined
 * @throws {TypeError} caso entrada não seja um número
 * @throws {RangeError} caso número seja menor que 0 ou maior que 999
 * @returns {boolean} caso verdadeiro retorna true, caso contrário false
 */
function propriedade3025(n) {

    if (n == null || n == undefined) {
        throw new Error("Entrada não pode ser null ou undefined");
    }
    if (typeof n !== 'number') {
        throw new TypeError("n precisa ser um número");
    }
    if (n < 0 || n > 9999) {
        throw new RangeError("Intervalo de n inválido");
    }

    const i = Math.floor(n / 100);
    const j = Math.floor(n % 100);

    return (i + j) * (i + j) == n;
}

File: aula-01/java/Ex01.java
public class Ex01 {

    public static void main(String[] args) {
		System.out.println(propriedade3025(3025));
    }
    
	public static boolean propriedade3025(int n) {
        
        if (n < 0) {
            throw new IllegalArgumentException("Intervalo de n inválido");
        }

        if (n > 9999) {
            throw new IllegalArgumentException("Intervalo de n inválido");
        }
        
		int i = n / 100;
		int j = n % 100;
		return (i + j) * (i + j) == n;
		
    }
			
}
